h2/src/main/org/h2/engine/Database.java

```diff
@@ -227,7 +227,7 @@ public Database(ConnectionInfo ci, String cipher) {
         this.filePasswordHash = ci.getFilePasswordHash();
         this.databaseName = databaseName;
         this.databaseShortName = parseDatabaseShortName();
-        this.maxLengthInplaceLob = Constants.DEFAULT_MAX_LENGTH_INPLACE_LOB;
+        this.maxLengthInplaceLob = persistent ? Constants.DEFAULT_MAX_LENGTH_INPLACE_LOB : Integer.MAX_VALUE;
```

Comment:
`new byte[Integer.MAX_VALUE` throws `java.lang.OutOfMemoryError: Requested array size exceeds VM limit`.

At least in some JVMs maximum length of `byte[]` is `Integer.MAX_VALUE - 2`, but in sources of Java `Integer.MAX_VALUE - 8 ` is used in `ByteArrayOutputStream` and other places, because actual limit may differ between JVMs.

Reply:
Why do you think that that new byte[Integer.MAX_VALUE] will ever be executed? This is just a threshold value below which LOBs will be allocated in memory, but allocation will be done with actual size. In my mind, if someone decided to have in-memory database with LOBs, at least they have to be of some reasonable sizes. How many 2Tb LOBs will fit into JVMheap anyway? But if you think it matters, so be it, Integer.MAX_VALUE - 8